src/useCases/user/create/create-user.usecase.ts

```diff
@@ -1,48 +1,69 @@
 import { AppError, Report, StatusCode } from '@expressots/core'
 import { provide } from 'inversify-binding-decorators'
-import {
-    ICreateUserRequestDTO,
-    ICreateUserResponseDTO,
-} from './create-user.dto'
+import { CreateUserRequestDTO, CreateUserResponseDTO } from './create-user.dto'
 import { UserRepository } from '@repositories/user/user.repository'
-import { User } from '@entities/user.entity'
+// import { User } from '@entities/user.entity'
+import { genSaltSync, hashSync } from 'bcrypt'
+import { Prisma } from '@prisma/client'
 
 @provide(CreateUserUseCase)
 class CreateUserUseCase {
     constructor(private userRepository: UserRepository) { }
 
-    execute(data: ICreateUserRequestDTO): ICreateUserResponseDTO | null {
+    async execute(data: CreateUserRequestDTO): Promise<CreateUserResponseDTO> {
         try {
-            // const { name, email } = data;
+            const salt = genSaltSync(10)
+            const hashedPassword = hashSync(data.user.password, salt)
 
+            await this.userRepository.create({
+                email: data.user.email,
+                password: hashedPassword,
+                username: data.user.username,
+            })
+
+            console.log('||||||||||||||||||||||||||||||||||')
+
+            // const user: User | null = null
```

Comment:
Remover console e comentarios inuteis
```suggestion
            })
```